### PR TITLE
My code review suggestions.

### PR DESCRIPTION
**Pull request by mcquadmj117**
### Analysis of the Program

**Description of the Project**:
This is an 'trip planning' application that allows a user to view and select events based off of their selected interests and location. The applications does this by allowing a user to select, save and delete these events at "bucket list items" and organize them in a way that helps the user plan their trip smoothly. 

**Was the program available in UC Github on time?**: Yes

**Is the program documented /commented well enough for you to understand?**: Yes

**Does the program compile?**: Yes

**Rationale behind my changes**:
The rationale behind my changes is mostly based off of the recommended coding conventions from Kotlinlang.org. These conventions make the code much easier to manage and read. This goes a long way in maintaining code and will aid greatly to the development of a project when working with others who will be adding to and reading your code.

**Links to my commits on my groups repo**
1. https://github.com/erswagner7/PuzzlePic/commit/43ce6acf86de7a5af9dd65f02494b20d9ee41094
2. https://github.com/erswagner7/PuzzlePic/commit/e7904812a0b5a9502359aa958cf968e3b9b4e5ca
3. https://github.com/erswagner7/PuzzlePic/commit/0a6e2347928f84867ee9e148faa57316f5a64533

**Technical Concepts Learned
What I learned from this code review**:
- I learned about "Trailing Commas'  and the benefits of using the convention.
Benefits included: 1. making version control cleaner 2. easier time adding or reordering elements in the future 3. simplifying code generation.(Learned that the convention is optional but recommended at declaration and up to discretion at call site )
- I learned about naming conventions for functions, properties and local variables in Kotlin. (In this case, the camel case naming convention should be used for functions, properties, and local variables.)
- I learned about formatting function signatures and function expression bodies, as well as when to use specific syntaxes and indentations. (In this case I learned that functions with longer signatures should indent its arguments after their commas. 
_Ex_.
override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,) =

I also learned that expression bodies should be used for functions consisting of a single expression (in this case 4 spaces should be used after an indent from the '='.)
_Ex_.
        savedInstanceState: Bundle?,) =
            inflater.inflate(R.layout.main_fragment, container, false,)

- I learned about the appropriate way to order the contents of a class.(In this case I learned that companion objects should be placed at the bottom of the Kotlin class.)
- I learned how to create an ArrayAdapter using the string array and a default spinner layout
_Ex_.   ArrayAdapter.createFromResource(
            this,
            R.array.interests_array,
            android.R.layout.simple_spinner_item,
        ).also { adapter ->
            adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
            spinner.adapter = adapter
        }

### CODE REVIEW

File: app/src/main/java/com/bucketlist/MainActivity.kt [hunk 1]
```diff
@@ -13,15 +13,15 @@ class MainActivity : AppCompatActivity() {
         setContentView(R.layout.main_activity)
         if (savedInstanceState == null) {
             supportFragmentManager.beginTransaction()
-                .replace(R.id.container, MainFragment.newInstance())
+                .replace(R.id.container, MainFragment.newInstance(),)
```
Comment (mcquadmj117) [inline app/src/main/java/com/bucketlist/MainActivity.kt:16]: Adding trailing comma as recommended here: https://kotlinlang.org/docs/coding-conventions.html#trailing-commas

File: app/src/main/java/com/bucketlist/MainActivity.kt [hunk 2]
```diff
@@ -13,15 +13,15 @@ class MainActivity : AppCompatActivity() {
...
                 .commitNow()
         }
         val spinner: Spinner = findViewById(R.id.interests_spinner)
 // Create an ArrayAdapter using the string array and a default spinner layout
         ArrayAdapter.createFromResource(
             this,
             R.array.interests_array,
-            android.R.layout.simple_spinner_item
+            android.R.layout.simple_spinner_item,
```
Comment (mcquadmj117) [inline app/src/main/java/com/bucketlist/MainActivity.kt:24]: Adding trailing comma as recommended here: https://kotlinlang.org/docs/coding-conventions.html#trailing-commas

File: app/src/main/java/com/bucketlist/dao/ILocalBucketListItemDAO.kt [hunk 3]
```diff
@@ -10,11 +10,11 @@ interface ILocalBucketListItemDAO {
     fun getAll()  : LiveData<List<BucketListItem>>
 
     @Insert(onConflict = OnConflictStrategy.REPLACE)
-    fun insertAll(bucketlistitems: BucketListItem)
+    fun insertAll(bucketListItems: BucketListItem)
```
Comment (mcquadmj117) [inline app/src/main/java/com/bucketlist/dao/ILocalBucketListItemDAO.kt:13]: changing naming convention to use camel case as recommended here: https://kotlinlang.org/docs/coding-conventions.html#function-names

File: app/src/main/java/com/bucketlist/dto/BucketListItem.kt [hunk 4]
```diff
@@ -4,7 +4,7 @@ import androidx.room.Entity
 import androidx.room.PrimaryKey
 
 @Entity(tableName="bucketlistitem")
-data class BucketListItem(var name: String = "", @PrimaryKey var itemID: Int = 0) {
+data class BucketListItem(var name: String = "", @PrimaryKey var itemID: Int = 0,) {
```
Comment (mcquadmj117) [inline app/src/main/java/com/bucketlist/dto/BucketListItem.kt:7]: Adding trailing comma as recommended here: https://kotlinlang.org/docs/coding-conventions.html#trailing-commas

File: app/src/main/java/com/bucketlist/ui/main/MainFragment.kt [hunk 5]
```diff
@@ -10,25 +10,25 @@ import com.bucketlist.R
...
+            inflater.inflate(R.layout.main_fragment, container, false,)
+
 
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
         // TODO: Use the ViewModel
     }
 
+    companion object {
```
Comment (mcquadmj117) [inline app/src/main/java/com/bucketlist/ui/main/MainFragment.kt:28]: Moving companion object to bottom of the class as recommended here: https://kotlinlang.org/docs/coding-conventions.html#class-layout

File: app/src/main/java/com/bucketlist/ui/main/MainViewModel.kt [hunk 6]
```diff
@@ -9,9 +9,9 @@ import androidx.lifecycle.viewModelScope
...
 
 class MainViewModel : ViewModel() {
-    private var _bucketlistitems: MutableLiveData<ArrayList<BucketListItem>> = MutableLiveData<ArrayList<BucketListItem>>()
+    private var _bucketListItems: MutableLiveData<ArrayList<BucketListItem>> = MutableLiveData<ArrayList<BucketListItem>>()
 
     internal var bucketlistitems:MutableLiveData<ArrayList<BucketListItem>>
-        get () {return _bucketlistitems}
-        set (value) {_bucketlistitems = value}
+        get () {return _bucketListItems}
+        set (value) {_bucketListItems = value}
```
Comment (mcquadmj117) [inline app/src/main/java/com/bucketlist/ui/main/MainViewModel.kt:16]: changing naming convention to use camel case as recommended here: https://kotlinlang.org/docs/coding-conventions.html#function-names

File: app/src/main/java/com/bucketlist/MainActivity.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/bucketlist/MainActivity.kt:16]: Hmmm, this is interesting.  I appreciate the best practice link, as I didn't realize this was an acceptable convention.

File: app/src/main/java/com/bucketlist/dao/ILocalBucketListItemDAO.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/com/bucketlist/dao/ILocalBucketListItemDAO.kt:13]: Good catch.

**Comment by discospiff** [comment]
Several good suggestions in this review.  I recommend merging this branch.

**Pull request merged**